samcli/commands/test_runner/init/cli.py

```diff
@@ -0,0 +1,199 @@
+""",
+    default=DEFAULT_RESOURCE_ARN_MAP_FILE_NAME,
+)
+@click.option(
+    "--runtime",
+    required=False,
+    type=click.Choice(SUPPORTED_RUNTIMES, case_sensitive=False),
+    help=f"Specify the runtime with which to run tests.",
+    default="python3.8",
+)
+@click.option(
+    "--image-uri",
+    required=True,
+    type=str,
+    help="THIS IS A TEMPORARY OPTION. Supply an image URI for Fargate to create a container to run tests.",
+)
+@pass_context
+def cli(
+    ctx: Any,
+    tag_key: Optional[str],
+    tag_value: Optional[str],
+    template_name: str,
+    env_file: str,
+    runtime: str,
+    image_uri: str,
+) -> None:
+    """
+    `sam test-runner init` command entry point
+    """
+
+    do_cli(
+        ctx=ctx,
+        tag_key=tag_key,
+        tag_value=tag_value,
+        template_name=template_name,
+        env_file=env_file,
+        runtime=runtime,
+        image_uri=image_uri,
+    )
+
+
+def do_cli(
+    ctx: Any,
+    tag_key: Optional[str],
+    tag_value: Optional[str],
+    template_name: str,
+    env_file: str,
+    runtime: str,
+    image_uri: str,
+) -> None:
+    """
+    implementation of the `sam test-runner init` command
+    """
+
+    # TODO: Merge these two options into one to follow AWS CLI (if we can get access to their parser)
+    if tag_key and not tag_value:
+        LOG.info(COLOR.red("Specified tag key without a tag value."))
+        return
+
+    if tag_value and not tag_key:
+        LOG.info(COLOR.red("Specified tag value without a tag key."))
+        return
+
+    if tag_value and tag_key:
+        tag_filters = [{"Key": tag_key, "Values": [tag_value]}]
+    else:
+        tag_filters = None
+
+    # TODO: When ready, public ECR repositories will be created holding images capable of running tests for each available runtime.
+    #       Until ready for public images, image URIs are supplied by option
+    # image_uri = get_image_uri(runtime)
+
+    if tag_filters is not None:
+        boto_client_provider = get_boto_client_provider_with_config(region=ctx.region, profile=ctx.profile)
+        resource_arn_list = query_tagging_api(tag_filters, boto_client_provider)
+
+        if not resource_arn_list:
+            raise NoResourcesMatchGivenTagException(
+                f"Given tags `Key={tag_key},Value={tag_value}` do not match any resources, were they entered incorrectly?"
+            )
+
+        template_generator = FargateRunnerCFNTemplateGenerator(resource_arn_list)
+        arn_map_generator = FargateRunnerArnMapGenerator()
+
+        resource_arn_map_yaml_string = arn_map_generator.generate_env_vars_yaml_string(resource_arn_list)
+
+        write_file(env_file, resource_arn_map_yaml_string)
+
+        LOG.info(
+            COLOR.green(
+                f"✓ Sucessfully generated an environment variable specification file `test_runner_arn_map.yaml`.\n"
```

Comment:
What is it used for?

Reply:
I put the explanation in a separate LOG statement below, just to use a different color

https://github.com/aws/aws-sam-cli/blob/0f34f6218a4a6fba36c82e7a885e32a55c01f92d/samcli/commands/test_runner/init/cli.py#L172-L178

Reply:
I still don't understand what it's used for. What uses the ARNs?

Reply:
Oh sorry, the ARNs can be used to access resources within customer tests. E.g. If a customer wanted to invoke a lambda in their integration test, they can do so by specifying the ARN. Having the arns exist as environment variables within the Fargate container allows the customer to write tests around these variables instead of copy pasting ARNs.

For example, test code can look something like this:

```
ENGLISH_LAMBDA_ARN = os.getenv('ENGLISH_LAMBDA')
def test_english_lambda():
    lambdaClient = boto3.client('lambda')
    response = lambdaClient.invoke(FunctionName=ENGLISH_LAMBDA_ARN)
    ...
```